framework/src/timeintegrators/ExplicitRK2.C

```diff
@@ -25,7 +25,8 @@ ExplicitRK2::validParams()
 ExplicitRK2::ExplicitRK2(const InputParameters & parameters)
   : TimeIntegrator(parameters),
     _stage(1),
-    _residual_old(_nl.addVector("residual_old", false, GHOSTED))
+    _residual_old(_nl.addVector("residual_old", false, GHOSTED)),
```

Comment:
Could we use a parallel vector instead of a ghosted vector for residual?

Reply:
Good point, we possibly can I guess.